specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2022-08-03/communityGallery.json

```diff
@@ -0,0 +1,609 @@
+            "schema": {
+              "$ref": "../../../common-types/v1/common.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List community gallery images.": {
+            "$ref": "./examples/communityGalleryExamples/CommunityGalleryImage_List.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/communityGalleries/{publicGalleryName}/images/{galleryImageName}/versions": {
+      "get": {
+        "tags": [
+          "CommunityGalleryImageVersions"
+        ],
+        "operationId": "CommunityGalleryImageVersions_List",
+        "description": "List community gallery image versions inside an image.",
+        "parameters": [
+          {
+            "$ref": "../../../common-types/v1/common.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "./galleryRPCommon.json#/parameters/LocationNameParameter"
+          },
+          {
+            "$ref": "../../../common-types/v1/common.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/PublicGalleryNameParameter"
+          },
+          {
+            "$ref": "#/parameters/GalleryImageNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/CommunityGalleryImageVersionList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common-types/v1/common.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List community gallery image versions.": {
+            "$ref": "./examples/communityGalleryExamples/CommunityGalleryImageVersion_List.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "PirCommunityGalleryResource": {
+      "properties": {
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource name"
+        },
+        "location": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource location"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type"
+        },
+        "identifier": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/CommunityGalleryIdentifier"
+        }
+      },
+      "description": "Base information about the community gallery resource in pir."
+    },
+    "CommunityGalleryArtifactTags": {
+      "type": "object",
+      "additionalProperties": {
```

Comment:
>additionalProperties [](http://example.com/codeflow?start=7&length=20)

additionalProperties pattern is highly discouraged by ARM, since it's hard for clients to use. Generally, this pattern should only be used when needed and the object properties are dynamic, unknown, or user defined. Ref: [https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi032-only-use-additionalproperties-when-the-object-properties-are-dynamic-unknown-or-user-defined](https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi032-only-use-additionalproperties-when-the-object-properties-are-dynamic-unknown-or-user-defined).

Reply:
I am using additionalProperties just because I saw another dictionary is using it. This is proxy-only resource, and we want to add 'artifactTags' on it, in our api model, it is like this: https://msazure.visualstudio.com/One/_git/Compute-CPlat-PIR?path=/src/CRP-PIR/ArtifactService/API/PrivateGallery/PrivateGalleryModel.cs&version=GBdevelop&line=1020&lineEnd=1020&lineStartColumn=1&lineEndColumn=76&lineStyle=plain&_a=contents

Do you have any suggestion about how to make it right? Thanks

Reply:
And what would schematizing this object look like?

Reply:
if the question is how to schematize, that would involve replacing it with a proper list of expected properties.

if the question is whether its ok to use additionalProperties in the context of collection of arbitrary user-provided, metadata, I think thats ok

Reply:
If this is indeed a tracked resource as it seems, then ARM supports tags as a top level property in the envelope. Please use that instead.

Reply:
This is proxy-only resource, not a tracked resource  @rkmanda

Reply:
@kangsun-ctrl You can declare a 'tags' property as additional on proxy resources too. They just won't be fully integrated fully with tags search etc, because that only works automatically for tracked resources. You might be able to integrate it with use of ARN.